src/main/kotlin/dev/ocpd/jsensible/rules/jpa/JpaRules.kt

```diff
@@ -28,4 +32,10 @@ object JpaRules {
     fun noEagerFetch(): ArchRule =
         noMembers().should(useEagerFetch())
             .because("no property should be fetched eagerly by default")
+
+    fun noNullableOrOptional(): ArchRule =
```

Comment:
```suggestion
    fun noMisalignedNullability(): ArchRule =
```
I think you got the implementation right but the naming is a bit confusing across the board. As we do have `Optional<T>` in JDK so naming it this way is a bit ambiguous.